node-js-basics-101/node_modules/asynquence/asq.src.js

```diff
@@ -0,0 +1,884 @@
+/*! asynquence
+    v0.10.0 (c) Kyle Simpson
+    MIT License: http://getify.mit-license.org
+*/
+
+(function UMD(name,context,definition){
+	if (typeof define === "function" && define.amd) { define(definition); }
+	else if (typeof module !== "undefined" && module.exports) { module.exports = definition(); }
+	else { context[name] = definition(name,context); }
+})("ASQ",this,function DEF(name,context){
+	"use strict";
+
+	var cycle, scheduling_queue,
+		timer = (typeof setImmediate !== "undefined") ?
+			function $$timer(fn) { return setImmediate(fn); } :
+			setTimeout
+	;
+
+	// Note: using a queue instead of array for efficiency
+	function Queue() {
+		var first, last, item;
+
+		function Item(fn) {
+			this.fn = fn;
+			this.next = void 0;
+		}
+
+		return {
+			add: function $$add(fn) {
+				item = new Item(fn);
+				if (last) {
+					last.next = item;
+				}
+				else {
+					first = item;
+				}
+				last = item;
+				item = void 0;
+			},
+			drain: function $$drain() {
+				var f = first;
+				first = last = cycle = null;
+
+				while (f) {
+					f.fn();
+					f = f.next;
+				}
+			}
+		};
+	}
+
+	scheduling_queue = Queue();
+
+	function schedule(fn) {
+		scheduling_queue.add(fn);
+		if (!cycle) {
+			cycle = timer(scheduling_queue.drain);
+		}
+	}
+
+	function tapSequence(def) {
+		// temporary `trigger` which, if called before being replaced
+		// above, creates replacement proxy sequence with the
+		// success/error message(s) pre-injected
+		function trigger() {
+			def.seq = createSequence.apply(ø,arguments).defer();
+		}
+
+		// fail trigger
+		trigger.fail = function $$trigger$fail() {
+			var args = ARRAY_SLICE.call(arguments);
+			def.seq = createSequence(function $$create$sequence(done){
+				done.fail.apply(ø,args);
+			})
+			.defer();
+		};
+
+		// listen for signals from the sequence
+		def.seq
+		// note: cannot use `seq.pipe(trigger)` because we
+		// need to be able to update the shared closure
+		// to change `trigger`
+		.val(function $$val(){
+			trigger.apply(ø,arguments);
+			return ASQmessages.apply(ø,arguments);
+		})
+		.or(function $$or(){
+			trigger.fail.apply(ø,arguments);
+		});
+
+		// make a sequence to act as a proxy to the original
+		// sequence
+		def.seq = createSequence(function $$create$sequence(done){
+			// replace the temporary trigger (created below)
+			// with this proxy's trigger
+			trigger = done;
```

Comment:
__[jshint]__
* Reassignment of 'trigger', which is is a function. Use 'var' or 'let' to declare bindings that may change.

<!-- Comment by [SideCI](https://sideci.com) -->